modules/dreambooth/conversion.py

```diff
@@ -0,0 +1,1454 @@
+        _copy_linear(hf_layer.final_layer_norm, pt_layer[1][0])
+
+        # copy attn
+        _copy_attn_layer(hf_layer.self_attn, pt_layer[0][1])
+
+        # copy MLP
+        pt_mlp = pt_layer[1][1]
+        _copy_linear(hf_layer.fc1, pt_mlp.net[0][0])
+        _copy_linear(hf_layer.fc2, pt_mlp.net[2])
+
+    def _copy_layers(hf_layers, pt_layers):
+        for i, hf_layer in enumerate(hf_layers):
+            if i != 0:
+                i += i
+            pt_layer = pt_layers[i: i + 2]
+            _copy_layer(hf_layer, pt_layer)
+
+    hf_model = LDMBertModel(config).eval()
+
+    # copy  embeds
+    hf_model.model.embed_tokens.weight = checkpoint.transformer.token_emb.weight
+    hf_model.model.embed_positions.weight.data = checkpoint.transformer.pos_emb.emb.weight
+
+    # copy layer norm
+    _copy_linear(hf_model.model.layer_norm, checkpoint.transformer.norm)
+
+    # copy hidden layers
+    _copy_layers(hf_model.model.layers, checkpoint.transformer.attn_layers.layers)
+
+    _copy_linear(hf_model.to_logits, checkpoint.transformer.to_logits)
+
+    return hf_model
+
+
+def convert_ldm_clip_checkpoint(checkpoint):
+    text_model = CLIPTextModel.from_pretrained("openai/clip-vit-large-patch14")
+
+    keys = list(checkpoint.keys())
+
+    text_model_dict = {}
+
+    for key in keys:
+        if key.startswith("cond_stage_model.transformer"):
+            text_model_dict[key[len("cond_stage_model.transformer."):]] = checkpoint[key]
+
+    text_model.load_state_dict(text_model_dict)
+
+    return text_model
+
+
+def extract_checkpoint(new_model_name: str, checkpoint_path: str, scheduler_type="ddim"):
+    # Set up our base directory for the model and sanitize our file name
+    new_model_name = "".join(x for x in new_model_name if x.isalnum())
+    config_data = {
+        "name": new_model_name,
+        "scheduler": scheduler_type,
+        "src": checkpoint_path,
+        "total_steps": 0
+    }
+    new_model_dir = create_output_dir(new_model_name, config_data)
+    # Create folder for the 'extracted' diffusion model.
+    out_dir = os.path.join(new_model_dir, "stable-diffusion-v1-4")
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+
+    # Todo: What impact does using the 'og' model config if not using that model as our base?
+    original_config_file = load_file_from_url("https://raw.githubusercontent.com/CompVis/stable-diffusion/main"
+                                              "/configs/stable-diffusion/v1-inference.yaml", new_model_dir
+                                              )
+
+    original_config = OmegaConf.load(original_config_file)
+    # Is this right?
+    checkpoint_file = modules.sd_models.get_closet_checkpoint_match(checkpoint_path)
+    if checkpoint_file is None or not os.path.exists(checkpoint_file[0]):
+        print("Unable to find checkpoint file!")
+        return None, "Unable to find base checkpoint.", ""
+    checkpoint = torch.load(checkpoint_file[0])["state_dict"]
```

Comment:
Maybe we should add some checks like https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/1878:
```python
  if "state_dict" in checkpoint:
         checkpoint = checkpoint["state_dict"]
```